UX improvement for Google account card on Ads setup page

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Part of UX improvement for Google Ads setup. It's also related to #1065.

- Adjust the section wordings and apply the `<ConnectedGoogleAccountCard>` to the ads setup page.
- Add a new prop `helper` to `<AccountCard>` and `<ConnectedGoogleAccountCard>` for showing a helper text below card description.
- Add a new prop `hideAccountSwitch` to `<ConnectedGoogleAccountCard>` for hiding the account switch within card footer.
- Tweak base styles of AccountCard to fit with the visual design of UX improvements project

### Additional notes

- This PR only adjusts the Google account card. The adjustments of Google Ads card will be a separate PR.
- After adding the `helper` prop, we could split part of glued `description` to `helper`, and it would be another PR once the main development finished
    1. [ConnectGoogleAccountCard](https://github.com/woocommerce/google-listings-and-ads/blob/5d5aca2e384a516917b1525f15e8e7ac432a6f89/js/src/components/google-account-card/connect-google-account-card.js#L23-L47)
    2. [RequestFullAccessGoogleAccountCard](https://github.com/woocommerce/google-listings-and-ads/blob/5d5aca2e384a516917b1525f15e8e7ac432a6f89/js/src/components/google-account-card/request-full-access-google-account-card.js#L33-L56)
    3. Maybe some usages within the contact information pages.

### Screenshots:

#### 🖼️ . Adjusted Google account card on Ads setup page

![image](https://user-images.githubusercontent.com/17420811/140724507-38d67c4b-bc13-4dcb-9fab-5b0c03b5bc53.png)

#### 🖼️ . Google account card on onboarding setup page is the same as the original

![image](https://user-images.githubusercontent.com/17420811/140724581-09eae709-e30c-425d-9bfe-78e961b411ae.png)

### Detailed test instructions:

1. Go to Google Ads setup page.
2. The visual of Google account card should be close to the design on Figma.
3. Go to onboarding setup page.
4. The connected Google account card should show the same UI blocks within itself as before.

### Changelog entry

eason9487 committed:
Add a new prop `helper` to AccountCard and ConnectedGoogleAccountCard

eason9487 committed:
Add a new prop `hideAccountSwitch` to ConnectedGoogleAccountCard

eason9487 committed:
Tweak base styles of AccountCard to fit with visual design of UX improvements project

eason9487 committed:
Apply the ConnectedGoogleAccountCard to the ads setup page

eason9487 committed:
Adjust the section wordings and Google account card's layer on ads setup page

eason9487 commented on js/src/setup-ads/ads-stepper/setup-accounts/index.js:
```diff
@@ -42,20 +42,22 @@ const SetupAccounts = ( props ) => {
-						'google-listings-and-ads'
-					) }
-				/>
+				<VerticalGapLayout size="large">
```
💡  Since the Google Ads card will be merged into this section in subsequent PR, the `<VerticalGapLayout>` was added here in advance.

ecgan commented on js/src/components/account-card/index.scss:
```diff
@@ -9,16 +9,29 @@
+	&__title:not(:empty) {
+		margin-bottom: 4px;
+		color: $black;
+	}
```
❔ 💅  Instead of using `:not(:empty)`, I'm thinking it might be better to check and conditionally render `Subsection.Title` in React like below? Similar to how you are doing it with `helper`? 🤔 

```js
{ title && <Subsection.Title className="gla-account-card__title">{ title }</Subsection.Title>}
```

Then, in CSS side, we just need `&__title`, without the need of `:not(:empty)`.

eason9487 replied:
Changed by https://github.com/woocommerce/google-listings-and-ads/pull/1083/commits/7c3b1af05bf31db194c52b8c1211f528a9269cb4 along with reverting the relevant changes made in https://github.com/woocommerce/google-listings-and-ads/commit/4e78d459877f18479e7145dc48daede8501c034a#diff-4dd8e87e5d8f40bcf64e31a91732fa0af9f4f7e7cbef6b69b5f2a816169bced6R10-R12

ecgan commented on js/src/components/account-card/index.scss:
```diff
@@ -9,16 +9,29 @@
 	}
 
+	&__helper {
+		margin-top: 4px;
```
💅 An alternative to this would be to apply `margin-bottom: 4px` to `&__description:not(:last-child)`, if we want to use top-down flow with `margin-bottom`.

eason9487 replied:
Thanks for the suggestion! I think we can keep it as is since the `helper` is conditionally rendered.

ecgan reviewed (approved):
Tested and working as expected. Left two minor non-blocking comments below. LGTM 👍

eason9487 committed:
Use conditional rendering instead of :not(:empty) styling for the `title` in AccountCard.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1083#discussion_r748302683

eason9487 commented:
Proceed to merge since the error of PHP unit tests is not related to this PR. 

Ref: #1093
